runtime/near-vm-runner/src/tests/runtime_errors.rs

```diff
@@ -445,7 +426,7 @@ fn test_stack_instrumentation_protocol_upgrade() {
         .opaque_error()
         .expects(&[
             expect![[r#"
-                VMOutcome: balance 4 storage_usage 12 return data None burnt gas 6789985365 used gas 6789985365
+                VMOutcome: balance 4 storage_usage 12 return data None burnt gas 18136872021 used gas 18136872021
```

Comment:
Hm, why would the gas values change here? Is there an issue with the new way of keeping versions? (Which I wonder, what's the reason for changing it?)

Reply:
I believe this change is actually indicative of the previous way not correctly using the old stack limiting mechanism. You will notice that previously the gas was exactly the same before (the first `expect!` and after (the 2nd `expect!`) the `ProtocolFeature::CorrectStackLimit` feature, which makes no sense given the nature of this test. That said, I haven’t looked too deeply into why this exact test was behaving the way it did before…

Reply:
Ok, that's a reasonable explanation. :)

Maybe for final PR, it would make sense to do the refactor of the `TestBuilder` in a separate PR to make it obvious it has nothing to do with code changes outside the testing framework.